client/network/src/service.rs

```diff
@@ -665,6 +665,11 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkService<B, H> {
 	pub fn num_connected(&self) -> usize {
 		self.num_connected.load(Ordering::Relaxed)
 	}
+
+	/// You must call this when a new block is imported by the client.
+	pub fn on_block_imported(&self, header: B::Header, data: Vec<u8>, is_best: bool) {
```

Comment:
This should be announce_block

Reply:
It's not just about announcing the block. It also updates the state of `sync` so that it knows what's in the database.

Reply:
There is already a method announce block there: https://github.com/paritytech/substrate/blob/61c5d4b6e0339acccfc37f5dbfa15cde0e4183ac/client/network/src/service.rs#L552-L558

Reply:
Yeah I know. For whatever we decide, `default_announce_block` should be named accordingly.

Reply:
Fair point

Reply:
> There is already a method announce block there:
> 
> https://github.com/paritytech/substrate/blob/61c5d4b6e0339acccfc37f5dbfa15cde0e4183ac/client/network/src/service.rs#L552-L558

I just realized this. This makes the whole story here even more confusing. We just want to announce a block later, but that would not update `sync` and would break everything... If we continue with this new method here, I would propose to remove this `announce_block` function.

@tomaka any ideas?

Reply:
Removing that function without replacing the functionality _exactly_ will break GRANDPA.